internal/internal_task_pollers.go

```diff
@@ -767,8 +771,15 @@ func (wtp *workflowTaskPoller) poll(ctx context.Context) (interface{}, error) {
 	response, err := wtp.service.PollForDecisionTask(ctx, request, getYarpcCallOptions(wtp.featureFlags)...)
 	if err != nil {
 		retryable := isServiceTransientError(err)
+
 		if retryable {
-			wtp.metricsScope.Counter(metrics.DecisionPollTransientFailedCounter).Inc(1)
+			if target := (*s.ServiceBusyError)(nil); errors.As(err, &target) {
+				tagsMap := map[string]string{}
+				tagsMap[causeTag] = serviceBusy
+				wtp.metricsScope.Tagged(tagsMap).Counter(metrics.DecisionPollTransientFailedCounter).Inc(1)
```

Comment:
`Tagged(map[string]string{causeTag: serviceBusy})` for simplification

Reply:
+1 for init-with-data whenever possible.  it's a safer habit, e.g. detects collisions.